packages/parser/src/ethereum/index.ts

```diff
@@ -0,0 +1,180 @@
+import { BigNumber } from 'bignumber.js'
+import { ethers } from 'ethers'
+import { Tx } from '@shapeshiftoss/blockbook'
+import { Tx as ParseTx, TxSpecific as ParseTxSpecific, Token, TransferType, Transfer } from '../types'
+import { InternalTx, Network } from './types'
+import { getSigHash } from './utils'
+import * as multiSig from './multiSig'
+import * as thor from './thor'
+import * as uniV2 from './uniV2'
+import * as zrx from './zrx'
+
+export interface TransactionParserArgs {
+  network?: Network
+  midgardUrl: string
+  rpcUrl: string
+}
+
+export class TransactionParser {
+  network: Network
+
+  private thor: thor.Parser
+  private uniV2: uniV2.Parser
+  private zrx: zrx.Parser
+
+  constructor(args: TransactionParserArgs) {
+    const provider = new ethers.providers.JsonRpcProvider(args.rpcUrl)
+
+    this.network = args.network ?? Network.Mainnet
+
+    this.thor = new thor.Parser({ network: this.network, midgardUrl: args.midgardUrl })
+    this.uniV2 = new uniV2.Parser({ network: this.network, provider })
+    this.zrx = new zrx.Parser()
```

Comment:
I think it would be better to have a `registerParser` mechanism to decouple the `ethereum` parser from these specific implementations. If we think of this package as independently useful, it would be great for developers to be able to write arbitrary parsers they can register rather than relying on the package to include support before they can use it.

Reply:
Agreed that would be a nice improvement. I started this module as a pretty 1:1 swap out of the current ingester parser. We can start making issues to track enhancements as deemed necessary